tests/components/todoist/test_todo.py

```diff
@@ -0,0 +1,221 @@
+"""Unit tests for the Todoist todo platform."""
+from unittest.mock import AsyncMock
+
+import pytest
+
+from homeassistant.components.todo import DOMAIN as TODO_DOMAIN
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_component import async_update_entity
+
+from .conftest import PROJECT_ID, make_api_task
+
+
+@pytest.fixture(autouse=True)
+def platforms() -> list[Platform]:
+    """Fixture to specify platforms to test."""
+    return [Platform.TODO]
+
+
+@pytest.mark.parametrize(
+    ("tasks", "expected_state"),
+    [
+        ([], "0"),
+        ([make_api_task(id="12345", content="Soda", is_completed=False)], "1"),
+        ([make_api_task(id="12345", content="Soda", is_completed=True)], "0"),
+        (
+            [
+                make_api_task(id="12345", content="Milk", is_completed=False),
+                make_api_task(id="54321", content="Soda", is_completed=False),
+            ],
+            "2",
+        ),
+    ],
+)
+async def test_todo_item_state(
+    hass: HomeAssistant,
+    setup_integration: None,
+    expected_state: str,
+) -> None:
+    """Test for a To-do List entity state."""
+
+    await async_update_entity(hass, "todo.name")
```

Comment:
Note to reviewer: In hindsight, I think I should go back and request update on start, then redo these test setups.

--- but still think this could be ready for a first pass of review.